### PR TITLE
feat: update scaffold

### PR DESCRIPTION
**Pull request by leohxj**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,6 +1,18 @@
 {
   "$schema": "http://json.schemastore.org/prettierrc",
-  "printWidth": 80,
+  "printWidth": 100,
```
Comment (JounQin) [inline .prettierrc]: I don't understand. Why?

File: .prettierrc [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+  "trailingComma": "all",
+  "htmlWhitespaceSensitivity": "ignore",
+  "overrides": [
+    {
+      "files": [
+        ".*rc",
+        "*.json"
+      ],
+      "options": {
+        "parser": "json-stringify"
```
Comment (JounQin) [inline .prettierrc]: Some special files like `tsconfig.json`, `tslint.json` are `jsonc`, and should use parser `json`.

File: .prettierrc (hunk 1 above)
Comment (JounQin) [reply .prettierrc]: Besides, we should always use union prettier configuration.

File: config/env.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
 
 const rootPath = process.cwd();
+const pkg = require(`${rootPath}/package.json`);
 
 module.exports = {
   rootPath,
-  srcPath: `${rootPath}/src`,
-  distPath: `${rootPath}/dist`,
-  publicPath: `${rootPath}/public`
+  srcPath: path.resolve(rootPath, 'src'),
```
Comment (JounQin) [inline config/env.js]: `rootPath` can be omit here.

File: config/env.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+  distPath: path.resolve(rootPath, 'dist'),
+  publicPath: path.resolve(rootPath, 'public'),
+  libraryName: process.env.LIBRARY_NAME || pkg.name,
```
Comment (JounQin) [inline config/env.js]: I have no idea when should we use environment instead of package.json.

File: config/webpack/lib/webpack.config.base.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+module.exports = {
+  entry: env.srcPath,
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js', '.json'],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(ts|tsx)?$/,
```
Comment (JounQin) [inline config/webpack/lib/webpack.config.base.js]: Bad regexp.

`\.tsx?$/`

File: config/webpack/lib/webpack.config.base.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        exclude: /node_modules/,
+        use: ['babel-loader'],
```
Comment (JounQin) [inline config/webpack/lib/webpack.config.base.js]: Should we always enable `cacheDirectory`?

File: config/webpack/lib/webpack.config.base.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      },
+    ],
+  },
+  plugins: [new CaseSensitivePathsPlugin()],
```
Comment (JounQin) [inline config/webpack/lib/webpack.config.base.js]: ForkTsCheckerPlugin should be added.

File: config/webpack/lib/webpack.config.production.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  mode: 'production',
+  output: {
+    filename: `${env.libraryName}.production.js`,
+    path: path.resolve(env.distPath, 'umd'),
+    library: env.libraryName,
+    libraryTarget: 'umd',
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      'process.env': {
```
Comment (JounQin) [inline config/webpack/lib/webpack.config.production.js]: It is enabled by default, should be removed.

File: config/webpack/react/webpack.config.dev.js [hunk 9]
```diff
@@ -39,14 +38,22 @@ const developmentConfig = merge(baseConfig, {
           {
             loader: 'eslint-loader',
```
Comment (JounQin) [inline config/webpack/react/webpack.config.dev.js]: So you still love it.🤪

File: config/webpack/react/webpack.config.base.js [hunk 10]
```diff
@@ -57,47 +57,47 @@ module.exports = {
...
+            },
+          },
+        ],
+      },
+    ],
   },
   optimization: {
     runtimeChunk: {
-      name: (entrypoint) => `runtime.${entrypoint.name}`
+      name: entrypoint => `runtime.${entrypoint.name}`,
```
Comment (JounQin) [inline config/webpack/react/webpack.config.base.js]: Maybe `manifest/runtime` should be enough here, why should we care about that?

File: packages/yab-dev-tools-api/package.json [hunk 11]
```diff
@@ -3,15 +3,20 @@
...
+  "module": "dist/es/index.js",
+  "types": "dist/types/index.d.ts",
+  "unpkg": "dist/umd/yab-dev-tools-api.production.js",
+  "files": [
+    "dist"
+  ],
   "scripts": {
     "clean": "rimraf dist",
-    "build": "npm run clean && npm run build:cjs && npm run build:es && npm run build:types",
+    "build": "npm run clean && npm run build:cjs && npm run build:es && npm run build:umd && npm run build:umd:min",
```
Comment (JounQin) [inline packages/yab-dev-tools-api/package.json]: Use something like `npm-run-all` to simplify the command.

File: packages/yab-dev-tools-ui/tsconfig.json [hunk 12]
```diff
@@ -2,10 +2,14 @@
   "$schema": "http://json.schemastore.org/tsconfig",
```
Comment (JounQin) [inline packages/yab-dev-tools-ui/tsconfig.json]: redundant for VSCode.

File: config/webpack/react/webpack.config.base.js [hunk 13]
```diff
@@ -1,28 +1,28 @@
 'use strict';
```
Comment (JounQin) [inline config/webpack/react/webpack.config.base.js]: Again, why we prefer webpack over parcel here?

And why not use `.ts` always consistently?

**Comment by leohxj** [comment]
> SD, the appraisal is complete.

Thanks for my dear son, I will fix those this afternoon.

File: config/webpack/react/webpack.config.dev.js (hunk 9 above)
Comment (leohxj) [reply config/webpack/react/webpack.config.dev.js]: Only use `eslint-loader`, `fork-ts-checker-webpack-plugin` in dev mode(build react-app).
In production mode, often work with CI, there is a lint flow.

File: config/webpack/react/webpack.config.base.js (hunk 10 above)
Comment (leohxj) [reply config/webpack/react/webpack.config.base.js]: Zhuangbility.

**Pull request merged**